pkg/operation/shoot/shoot.go

```diff
@@ -193,12 +197,16 @@ func (s *Shoot) GetReplicas(wokenUp int) int {
 // ComputeAPIServerURL takes a boolean value identifying whether the component connecting to the API server
 // runs in the Seed cluster <runsInSeed>, and a boolean value <useInternalClusterDomain> which determines whether the
 // internal or the external cluster domain should be used.
-func (s *Shoot) ComputeAPIServerURL(runsInSeed, useInternalClusterDomain bool) string {
+func (s *Shoot) ComputeAPIServerURL(runsInSeed, useInternalClusterDomain bool, apiServerAddress string) string {
 	if runsInSeed {
 		return v1alpha1constants.DeploymentNameKubeAPIServer
 	}
 
-	if s.Info.Spec.DNS != nil && len(s.Info.Spec.DNS.Providers) > 0 && s.Info.Spec.DNS.Providers[0].Type != nil && *s.Info.Spec.DNS.Providers[0].Type == "unmanaged" {
+	if s.DisableDNS {
```

Comment:
Yes, actually, this taint should not be removed/added if there are shoots on this seed. I have thought about validation, though, this would require it to be in an admission plugin. I’ll build it in the resourcereferencemanager plugin for now, as this is the only good place atm. This should also address your following comment/observation.